### PR TITLE
Adding functionality for Project Cards

### PR DESCRIPTION
**Pull request by jpomfret**
Covers:
- List project cards
- Get a project card
- Create a project card
- Update a project card
- Delete a project card
- Move a project card

The tests don't run successfully without the Project code from PR #160, if you want I can rebase this branch with master once/if the core project code gets merged in?

### CODE REVIEW

File: GitHubProjectCards.ps1 [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+        if ($key -in ('Top', 'Bottom', 'After'))
+        {
+            if($PSBoundParameters[$key] -ne $false)
+            {
+                $paramsCount ++
+            }
+        }
+    }
+
+    if($paramsCount -ne 1)
```
Comment (HowardWolosky) [inline GitHubProjectCards.ps1:473]: You can drop the whole $paramsCount section above, and just replace this if like so:

```suggestion
    if (-not ($Top -xor $Bottom -xor ($After -gt 0)))
```

For more info on xor, [see here](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_logical_operators).

File: GitHubProjectCards.ps1 [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    {
+        $message = 'You must use one of the parameters Top, Bottom or After.'
+        Write-Log -Message $message -level Error
+        throw $message
+    }
+    elseif($Top)
```
Comment (HowardWolosky) [inline GitHubProjectCards.ps1:479]:
```suggestion
    elseif ($Top)
```

File: GitHubProjectCards.ps1 [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+    {
+        $position = 'top'
+    }
+    elseif($Bottom)
```
Comment (HowardWolosky) [inline GitHubProjectCards.ps1:483]:
```suggestion
    elseif ($Bottom)
```

File: GitHubProjectCards.ps1 [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+    {
+        $position = 'bottom'
+    }
+    else
+    {
+        $position = "after:$After"
+    }
+
+    $hashBody = @{
+        'position' = $Position
```
Comment (HowardWolosky) [inline GitHubProjectCards.ps1:493]:
```suggestion
        'position' = $position
```

File: GitHubProjectCards.ps1 [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+            if($PSBoundParameters[$key] -ne $false)
+            {
+                $paramsCount ++
+            }
+        }
+    }
+
+    if($paramsCount -ne 1)
+    {
+        $message = 'You must use one of the parameters Top, Bottom or After.'
```
Comment (HowardWolosky) [inline GitHubProjectCards.ps1:475]:
```suggestion
        $message = 'You must use one (and only one) of the parameters Top, Bottom or After.'
```

File: Tests/GitHubProjectCards.tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        Context 'Move column using after parameter' {
+            $null = Move-GitHubProjectCard -Card $cardTwo.id -After $card.id
+            $results = Get-GitHubProjectCard -Column $column.id
+
+            It 'Card now exists in new column' {
+                $results[1].note | Should be $defaultCardTwo
+            }
+        }
+
+        Context 'Move column to another column' {
```
Comment (HowardWolosky) [inline Tests/GitHubProjectCards.tests.ps1:130]:
```suggestion
        Context 'Move card to another column' {
```

File: Tests/GitHubProjectCards.tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    }
+
+    Describe 'Create Project Cards' -tag new {
+        Context 'Create project card with note' {
+            BeforeAll {
+                $card = @{id = 0}
+            }
+            AfterAll {
+                $null = Remove-GitHubProjectCard -Card $card.id -Confirm:$false
+                Remove-Variable card
```
Comment (HowardWolosky) [inline Tests/GitHubProjectCards.tests.ps1:147]:
```suggestion
                Remove-Variable -Name card
```

File: Tests/GitHubProjectCards.tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+            }
+        }
+
+        Context 'Create project card from issue' {
+            BeforeAll {
+                $card = @{id = 0}
+            }
+            AfterAll {
+                $null = Remove-GitHubProjectCard -Card $card.id -Confirm:$false
+                Remove-Variable card
```
Comment (HowardWolosky) [inline Tests/GitHubProjectCards.tests.ps1:168]:
```suggestion
                Remove-Variable -Name card
```

**Comment by HowardWolosky** [comment]
Can you resolve the merge conflict now in the .psd1 again (since a new conflict occurred now that the Columns work was merged in).  Once resolved and I've verified a successful test pass, I'll get this one merged in, along with your documentation update.  Then I'll rev the version number and publish out a new version.

Thanks for all your help!

**Comment by HowardWolosky** [comment]
After your merges, the psd1 is in that bad state again (every line changed).  Can you please try the same fix again as you did on the other PR?

**Comment by jpomfret** [comment]
Alright - should be good now...

**Comment by HowardWolosky** [comment]
This is now available publicly in [0.12.0](https://www.powershellgallery.com/packages/PowerShellForGitHub/0.12.0).  Thanks so much for your contribution. 🎆

**Comment by HowardWolosky** [comment]
FYI @jpomfret - In my upcoming pipeline enabling change, I'm going to be changing the signature of `New-GitHubProjectCard` a little bit.  Instead of having to specify `ContentId` and `ContentType` in order to associate an issue/pullrequest, users can directly either pass in the `IssueId` or `PullRequestId` (and the content type will be inferred based on which is provided).  ParameterSets will be used to prevent both being passed in.

**Comment by jpomfret** [comment]
Makes sense to me, thanks for the heads up

**Comment by jpomfret** [comment]
Hey @HowardWolosky - has your change for ContentId --> IssueId/PullRequesetId made it to the PowerShell Gallery? Downloaded 0.14 today but it doesn't have this change in, even though I see it in this repo for version 0.14.

(Am finally finishing up a blog post on creating project/issue/cards with PowerShell so wanted to make sure I was accurate 😄)

**Comment by HowardWolosky** [comment]
Hi @jpomfret -- [0.14.0 is from 5/30/2020](https://github.com/microsoft/PowerShellForGitHub/blob/master/CHANGELOG.md).  There are tags in `master` that will sync you back to the state of the code for any given release.  The change I discussed above _is_ in `master`, but not part of any official release yet.  There were a _bunch_ of breaking changes that have been happening, and I have been trying to get all of them into a single release in order to help avoid having a series of releases with breaking changes.

The PR pipeline is just about flushed through of all the breaking changes.  I can probably start working on the CHANGELOG update today so that I can release 0.15.0 within the next day or two.

**Comment by HowardWolosky** [comment]
@jpomfret -- Decided to do a little more work in the module to take advantage of all the breaking changes in the coming release.  I think `0.15.0` will be released by this weekend.

As an FYI, an additional breaking change for your changes is that the `Set-GitHub*` commands will now be silent by default unless `-PassThru` is specified.  See #276 for more info.

**Comment by jpomfret** [comment]
Sounds good! Sorry for the slow reply, looking forward to the new release. ☺️

**Pull request merged**